src/main/Main.hs

```diff
@@ -755,7 +755,14 @@ sdistCmd sdistOpts go =
             liftIO $ L.writeFile (toFilePath tarPath) tarBytes
             checkSDistTarball sdistOpts tarPath
             prettyInfoL [flow "Wrote sdist tarball to", display tarPath]
+            unless (null $ sdoptsTarPath sdistOpts) (copyTarToTarPath tarPath tarName)
             when (sdoptsSign sdistOpts) (void $ Sig.sign (sdoptsSignServerUrl sdistOpts) tarPath)
+        where
+          copyTarToTarPath tarPath tarName = liftIO $ do
+            targetDir <- resolveDir' $ sdoptsTarPath sdistOpts
```

Comment:
I believe this will fail is the directory does not already exist (though I haven't tested it locally). How about:

I think you'll want to create the directory first with `D.createDirectoryIfMissing` and then use `resolveDir'`. Alternatively: in this case you may want to stick to `FilePath` values and use `</>` from `System.FilePath`.